src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -342,4 +362,100 @@ private String getCustomEnvironmentProperty(String parameterizedPropertyName, St
         }
 
     }
+
+    @DeleteMapping(REQUEST_PAYMENTS_POSTEPAY_REQUEST_ID)
+    public ResponseEntity<PostePayRefundResponse> refundPostePayPayment(@PathVariable String requestId,
+                                                                        @RequestHeader(required = false, value = MDC_FIELDS) String mdcFields) throws RestApiException {
+        setMdcFields(mdcFields);
+        log.info("START - refund PostePay request for paymentRequestEntity with guid: " + requestId);
+
+        PaymentRequestEntity requestEntity = paymentRequestRepository.findByGuid(requestId);
+
+        if (Objects.isNull(requestEntity) || !REQUEST_PAYMENTS_POSTEPAY.equals(requestEntity.getRequestEndpoint())) {
+            log.error("No PostePay response entity object has been found with guid " + requestId);
+            return createPostePayRefundResponse(requestId, null, null, ExceptionsEnum.TRANSACTION_NOT_FOUND);
+        }
+
+        if (requestEntity.getIsRefund()) {
+            log.info("Transaction with id " + requestEntity.getIdTransaction() + " has already be refunded");
+            return createPostePayRefundResponse(requestId, requestEntity.getCorrelationId(), null, ExceptionsEnum.TRANSACTION_ALREADY_REFUNDED);
+        }
+
+        // step 1. acquiring payment details PostePay side in order to validate the deletion request
+        DetailsPaymentRequest detailsPaymentRequest = createDetailPaymentRequest(requestEntity);
+
+        boolean isAuthCodeEmpty = StringUtils.isEmpty(requestEntity.getAuthorizationCode());
+        boolean checkDetail = true;
+
+        if (isAuthCodeEmpty) {
+            try {
+                log.info("Starting check Details for detailsPaymentRequest with guid: " + requestEntity.getGuid());
+                checkDetail = checkDetailStatus(detailsPaymentRequest);
+            } catch (Exception e) {
+                log.error("Generic Exception while checkDetailStatus DetailsPaymentRequest with paymentId: "
```

Comment:
Careful! here you only log the exception, but not set **checkDetail** variable to false.
This means that this way, the refund will always be requested whether an exception occurs

Reply:
This is the desired behaviour, the refund must proceed even if the check status does not respond or throws an exception